### PR TITLE
Media browser

### PR DESCRIPTION
**Pull request by zsarnett**
## Proposed change

6dba66c8bdcabcc9b3c7ad16e0ed1353

image

image

## Type of change

- [x] New feature (thank you!)

## Example configuration

```js
<ha-media-player-browse
  .hass=${this.hass}
  entityId="media_player.plex_plex_web_chrome"
  action="pick"
  @media-picked=${(ev) => console.log(ev.detail)}
></ha-media-player-browse>
```

## Additional information

- This PR is related to backend PR: https://github.com/home-assistant/core/pull/35590
- Link to documentation pull request: TODO

## Checklist

- [x] Add this to a dialog somewhere
- [ ] Documentation added/updated for [www.home-assistant.io][docs-repository]

### CODE REVIEW

**Comment by bramkragten** [comment]
So we don't actually have a place where we use this right now? Let's add it to the media player card to start with if the media player supports it.

File: src/components/media-player/dialog-media-player-browse.ts [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+  private _closeDialog() {
+    this._params = undefined;
+  }
+
+  private _mediaPicked(ev: HASSDomEvent<MediaPickedEvent>): void {
+    const mediaPicked = ev.detail;
+
+    fireEvent(this, "media-picked", mediaPicked);
+    this._closeDialog();
+  }
```
Comment (bramkragten) [inline src/components/media-player/dialog-media-player-browse.ts:91]: This doesn't work, as the dialog is not a child of the element that called it, this should be a callback that is passed in the dialog params.

File: src/components/media-player/ha-media-player-browse.ts [hunk 2]
```diff
@@ -0,0 +1,699 @@
...
+              <ha-data-table
+                .columns=${this._columns}
+                .data=${mostRecentItem.children.map((child) => {
+                  return { icon: this.action, ...child };
+                }) as DataTableRowData[]}
+                .autoHeight=${true}
+                .showHeader=${false}
+                .id=${"media_content_id"}
+                @row-click=${this._rowClicked}
+              ></ha-data-table>
```
Comment (bramkragten) [inline src/components/media-player/ha-media-player-browse.ts:287]: Why do we use a `data-tabel` here? We don't use any of it's functions, can't we better use a `mwc-list`?

Comment (bramkragten) [reply src/components/media-player/ha-media-player-browse.ts:287]:
```ts
              <mwc-list>
                ${mostRecentItem.children.map(
                  (child) => html`<mwc-list-item
                    @click=${this._actionClicked}
                    .item=${child}
                    graphic="icon"
                  >
                    <span>${child.title}</span>
                    <ha-svg-icon
                      slot="graphic"
                      .label=${this.hass.localize(
                        `ui.components.media-browser.${this.action}-media`
                      )}
                      .path=${this.action === "play" ? mdiPlay : mdiPlus}
                    ></ha-svg-icon
                  ></mwc-list-item>`
                )}
              </mwc-list>
```

Comment (bramkragten) [reply src/components/media-player/ha-media-player-browse.ts:287]: (and if you want the same styling, use `<li divider role="separator"></li>` and some css on the `mwc-list`)

File: src/dialogs/more-info/controls/more-info-media_player.ts [hunk 3]
```diff
@@ -62,15 +62,29 @@ class MoreInfoMediaPlayer extends LitElement {
...
-                    .icon=${control.icon}
-                    @click=${this._handleClick}
-                  ></ha-icon-button>
-                `
-              )}
+              <div class="basic-controls">
+                ${controls!.map(
+                  (control) => html`
+                    <ha-icon-button
+                      action=${control.action}
```
Comment (bramkragten) [inline src/dialogs/more-info/controls/more-info-media_player.ts:69]: We should add titles here too in a future PR

Comment (zsarnett) [reply src/dialogs/more-info/controls/more-info-media_player.ts:69]: And to the Media card as well

**Pull request merged**